### ISSUE TITLE
could not find profile at requested time and selectors

### ISSUE THREAD

**Issue by rafaeljesus** [parca-dev/parca-agent#1060]
I've upgraded the parca agent to `v0.10.0-rc.0` and when clicking into any of the profiles it renders the `could not find profile at requested time and selectors` error as follows

<img width="1414" alt="Screenshot 2022-11-22 at 22 01 15" src="https://user-images.githubusercontent.com/1216425/203420323-ff186d9e-fe78-4c4b-9545-3660193fad82.png">

on top of that, the autocomplete is not "autocompleting"

- agent version: v0.10.0-rc.0
- server version: v0.14.0 (also tasted with v0.13.0 and the same behavior applies)

I'd appreciate any help, thanks in advance

**Comment by brancz** [comment]
We’ve had someone else report this. Could you share a few things: 

* what browser (and version) are you using?
* Do merge queries work?

**Comment by rafaeljesus** [comment]
> what browser (and version) are you using?

I tested this in Firefox=`106.0.3` and Chrome=`107.0.5304.110`, Safari=`15.5`(all most recent versions)

> Do merge queries work?

yes it does

**Comment by brancz** [comment]
Which operating system and version?

and does the same thing happen on https://demo.parca.dev/ ?

**Comment by Sylfrena** [comment]
@brancz and I troubleshooted this and found out that the following parca config seems to reproduce this:
```
object_storage:
  bucket:
    type: "FILESYSTEM"
    config:
      directory: "./data"

scrape_configs:
  - job_name: "default"
    scrape_interval: "3s"
    static_configs:
      - targets: [ '127.0.0.1:7070' ]
        labels:
          faulty_label: \x2d
```

We are handling the "\x2d" incorrectly (probably in the ui).

cc @manojVivek @monicawoj @yomete

**Comment by rafaeljesus** [comment]
> Which operating system and version?
> 
> and does the same thing happen on https://demo.parca.dev/ ?

1) I tested this in Mac BigSur and Arch Linux with the latest kernel
2) Default search works fine in demo.parca.dev, but autocomplete doesn't

**Comment by rafaeljesus** [comment]
> @brancz and I troubleshooted this and found out that the following parca config seems to reproduce this:
> 
> ```
> object_storage:
>   bucket:
>     type: "FILESYSTEM"
>     config:
>       directory: "./data"
> 
> scrape_configs:
>   - job_name: "default"
>     scrape_interval: "3s"
>     static_configs:
>       - targets: [ '127.0.0.1:7070' ]
>         labels:
>           faulty_label: \x2d
> ```
> 
> We are handling the "\x2d" incorrectly (probably in the ui).
> 
> cc @manojVivek @monicawoj @yomete

thanks for checking that, the only difference with my config is that server is not scraping anything, all I have is
```yaml
    object_storage:
      bucket:
        type: "FILESYSTEM"
        config:
          directory: "./tmp"
```

**Comment by brancz** [comment]
Ref on Prometheus side: https://github.com/prometheus/prometheus/issues/11629

tl;dr we need to handle escaping correctly in the frontend
